### ISSUE TITLE
Bug: RepositoryServices and Repositories have different list_and_count/count filter signatures

### ISSUE THREAD

**Issue by H00N24** [jolt-org/advanced-alchemy#107]
### Description

Hi,

First of all thanks, for the package. I've encountered a mypy error with filters as `list_and_count` and `count` methods in `*RepositoryService` ([*filters: FilterTypes](https://github.com/jolt-org/advanced-alchemy/blob/ba58db712db057d65e14b79930c39a3778e8c758/advanced_alchemy/service/_async.py#L179)) and `*Repository` ([*filters: FilterTypes | ColumnElement[bool]](https://github.com/jolt-org/advanced-alchemy/blob/ba58db712db057d65e14b79930c39a3778e8c758/advanced_alchemy/repository/_async.py#L763)) classes have different signatures.

Is there a specific reason to have a drift in signatures as they are only forwarding them? If not I can prepare a PR with a fix. Thanks. 

### MCVE

```python
filters: list[FilterTypes | ColumnElement[bool]] = []

repository = UserReposiotry(session=db_session)
repository.list_and_count(*filters) # OK
repository.count(*filters) # OK

service = UserService(session=db_session)
service.list_and_count(*filters) # Error, expects: *filters: FilterTypes
service.count(*filters) # Error, expects: *filters: FilterTypes
```

### Jolt Project Version

v0.5.5

### Platform

- [X] Linux
- [X] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

---

## Funding
* If you would like to see an issue prioritized, make a pledge towards it!
* We receive the pledge once the issue is completed & verified

<a href="https://polar.sh/jolt-org/advanced-alchemy/issues/107">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/jolt-org/advanced-alchemy/issues/107/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/jolt-org/advanced-alchemy/issues/107/pledge.svg">
</picture>
</a>

**Comment by cofin** [comment]
@H00N24, thanks for point this out.  This looks like an oversight.  Based on my quick review, this should b there.  If you'd like to make this change, please go for it.  it would be greatly appreciated.

BTW, if you just make the change on the `_async.py` file and run `make lint`, it will use `unasyncd` to replicate the change to `_sync.py`.  This allows us to only change the async repo/services and have the sync modules generated for us.

**Comment by cofin** [comment]
I hope you don't mind, but I addressed this today while I was wrapping up the encrypted fields PR.

Resolved by #108
